### PR TITLE
Rewarded ad integration & README update

### PR DESCRIPTION
**Pull request by yunusemrepekguc**

### CODE REVIEW

File: huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomAdapter.kt [hunk 1]
```diff
@@ -7,42 +7,36 @@ import com.ironsource.mediationsdk.adunit.adapter.BaseAdapter
...
-            Log.e(TAG, "Init Failed: $stacktrace")
+            Log.d(TAG, "Adapter - init() - Init failed. $e")
+
             listener?.onInitFailed(AdapterErrors.ADAPTER_ERROR_INTERNAL, e.toString())
         }
     }
 
     override fun getNetworkSDKVersion(): String {
-        Log.d(TAG, "getNetworkSDKVersion() = ${HwAds.getSDKVersion()}")
+        Log.e(TAG, "Adapter - getNetworkSDKVersion ${HwAds.getSDKVersion()}")
```
Comment (oguzozsoy) [inline huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomAdapter.kt]: I think this log type should not be an error.

File: huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomAdapter.kt [hunk 2]
```diff
@@ -7,42 +7,36 @@ import com.ironsource.mediationsdk.adunit.adapter.BaseAdapter
...
         return HwAds.getSDKVersion()
     }
 
     override fun getAdapterVersion(): String {
-        Log.d(TAG, "getAdapterVersion() = ${getVersion()}")
+        Log.e(TAG, "Adapter - getAdapterVersion ${getVersion()}")
```
Comment (oguzozsoy) [inline huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomAdapter.kt]: I think this log type should not be an error.

File: huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomAdapter.kt [hunk 3]
```diff
@@ -7,42 +7,36 @@ import com.ironsource.mediationsdk.adunit.adapter.BaseAdapter
...
 import com.ironsource.mediationsdk.adunit.adapter.utility.AdapterErrors
-import java.io.PrintWriter
-import java.io.StringWriter
 
 class HuaweiAdsAdapterCustomAdapter : BaseAdapter() {
 
-    private val TAG = HuaweiAdsAdapterCustomAdapter::class.simpleName
+    private fun getVersion() = ADAPTER_VERSION
+    private var ADAPTER_VERSION = "1.0"
+    private var TAG = HuaweiAdsAdapterCustomAdapter::class.simpleName
```
Comment (oguzozsoy) [inline huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomAdapter.kt]: I think it is better to define TAG as const

File: huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomInterstitial.kt [hunk 4]
```diff
@@ -10,33 +10,34 @@ import com.ironsource.mediationsdk.adunit.adapter.utility.AdData
...
-import java.io.StringWriter
-import java.io.PrintWriter
 
 class HuaweiAdsAdapterCustomInterstitial(networkSettings: NetworkSettings) :
-        BaseInterstitial<HuaweiAdsAdapterCustomAdapter>(networkSettings) {
+    BaseInterstitial<HuaweiAdsAdapterCustomAdapter>(networkSettings) {
     private var mInterstitialAd: InterstitialAd? = null
     private var mHuaweiAdsAdapterConfiguration = HuaweiAdsAdapterCustomAdapter()
-    private val TAG = HuaweiAdsAdapterCustomInterstitial::class.simpleName
+    var TAG = HuaweiAdsAdapterCustomInterstitial::class.simpleName
```
Comment (oguzozsoy) [inline huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomInterstitial.kt]: It is better to define TAG as const

File: huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomInterstitial.kt [hunk 5]
```diff
@@ -10,33 +10,34 @@ import com.ironsource.mediationsdk.adunit.adapter.utility.AdData
...
 
+    private var AD_UNIT_ID = "adunitid"
+    private var TAG_FOR_CHILD_PROTECTION = "tagforchildprotection"
```
Comment (oguzozsoy) [inline huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomInterstitial.kt]: it should be const val

File: huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomInterstitial.kt [hunk 6]
```diff
@@ -10,33 +10,34 @@ import com.ironsource.mediationsdk.adunit.adapter.utility.AdData
...
+    private var TAG_UNDER_AGE_OF_PROMISE = "tagunderageofpromise"
```
Comment (oguzozsoy) [inline huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomInterstitial.kt]: it should be const val

File: huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomInterstitial.kt [hunk 7]
```diff
@@ -10,33 +10,34 @@ import com.ironsource.mediationsdk.adunit.adapter.utility.AdData
...
 
     override fun loadAd(adData: AdData, activity: Activity, listener: InterstitialAdListener) {
-        val instanceId = adData.getString("adunitid")
-        Log.i(TAG, "loadAd for instanceId: $instanceId")
+        val instanceId = adData.getString(AD_UNIT_ID)
+        Log.d(TAG, "Interstitial - loadAd() - for instanceId: $instanceId")
 
         if (instanceId == null) {
-            Log.e(TAG, "ad failed to load. instanceId is null")
+            Log.w(TAG, "Interstitial - loadAd() - instanceId is null")
```
Comment (oguzozsoy) [inline huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomInterstitial.kt]: this log type can be error

File: huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomRewardedVideo.kt [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+
+class HuaweiAdsAdapterCustomRewardedVideo(networkSettings: NetworkSettings) :
+    BaseRewardedVideo<HuaweiAdsAdapterCustomAdapter>(networkSettings) {
+
+    private var mRewardAd: RewardAd? = null
+    private var mHuaweiAdsAdapterConfiguration = HuaweiAdsAdapterCustomAdapter()
+    private var TAG = HuaweiAdsAdapterCustomInterstitial::class.simpleName
+
+    private var TAG_FOR_CHILD_PROTECTION = "tagforchildprotection"
```
Comment (oguzozsoy) [inline huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomRewardedVideo.kt]: it should be const val

**Review by oguzozsoy — COMMENTED** [review]
There is no big issue but I have written some suggestions.

**Comment by GenarC** [comment]
Because class names are acquired from javaClass library, it returns as "String?" type and it blocks const keyword usage. Therefore we couldn't use const, instead of private val.

Logs are updated

File: huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomAdapter.kt (hunk 1 above)
Comment (GenarC) [reply huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomAdapter.kt]: Log type is fixed

File: huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomAdapter.kt (hunk 2 above)
Comment (GenarC) [reply huaweiadsadapter/src/main/java/com/ironsource/adapters/custom/huaweiadsadapter/HuaweiAdsAdapterCustomAdapter.kt]: Log type is fixed

**Review by oguzozsoy — APPROVED** [review]
It is ok.

**Review by GenarC — APPROVED** [review]

**Pull request merged**